src/v2/Utils/Hooks/useMode.ts

```diff
@@ -0,0 +1,14 @@
+import { useState } from "react"
+
+/**
+ * This is basically just a wrapper around `useState` for intent signaling.
```

Comment:
Smart

Reply:
🤷 Its slightly silly, but it does feel-better code wise now